qtaf-core/src/main/java/de/qytera/qtaf/testng/context/QtafTestNGContext.java

```diff
@@ -98,6 +105,18 @@ public QtafTestNGContext initialize() {
         return this;
     }
 
+    /**
+     * Load class instance
+     * @param context   The current test context (pass 'this' for this argument)
+     * @param c         The desired class you want to create an instance of
+     * @return          Instance of the desired class
+     */
+    protected static Object loadClassInstance(IQtafTestContext context, Class<?> c) {
+        IQtafTestContext pageObject = (IQtafTestContext) injector.getInstance(c);
+        pageObject.setLogCollection(context.getLogCollection());
+        return pageObject;
+    }
+
```

Comment:
I feel like the following might be easier/cleaner code:

```java
protected static <T> T loadClassInstance(IQtafTestContext context, Class<T> c) {
    T instance = injector.getInstance(c);
    if (instance instanceof IQtafTestContext pageObject) {
        pageObject.setLogCollection(context.getLogCollection());
    }
    return instance;
}
```

This avoids casts on the callee side. It also prevents `ClassCastExceptions`. Unless you explicitly want them to occur to remind QTAF users that all page objects must extend `IQtafTestContext`, then feel free to keep the "unchecked" cast.